### PR TITLE
revise doc for making keyGenerateAlgorithm as a spring bean.

### LINKED ISSUE #5041: revise doc for making keyGenerateAlgorithm as a spring bean
for #1977
revise doc for making keyGenerateAlgorithm as a spring bean.

### PR DESCRIPTION
**Pull request by TaoZhiMLND**
Fixes #5041.

### CODE REVIEW

File: docs/document/content/manual/sharding-jdbc/configuration/config-java.cn.md [hunk 1]
```diff
@@ -285,15 +285,22 @@ ShardingStrategyConfiguration的实现类，用于配置不分片的策略。
...
-| ----------------- | ---------------------------- | ------------------------------------------------------------------------------ |
-| column            | String                       | 自增列名称                                                                      |
-| type              | String                       | 自增列值生成器类型，可自定义或选择内置类型：SNOWFLAKE/UUID/LEAF_SEGMENT/LEAF_SNOWFLAKE|
-| props             | Properties                   | 自增列值生成器的相关属性配置                                                      |  
+| *名称*                 | *数据类型*                    | *说明*                                                                         |
+| --------------------- | ---------------------------- | ------------------------------------------------------------------------------ |
+| column                | String                       | 自增列名称                                                                      |
+| keyGenerateAlgorithm  | KeyGenerateAlgorithm         | 自增列值生成算法                                                                 |
+
+#### KeyGenerateAlgorithm
```
Comment (cherrylzhao) [inline docs/document/content/manual/sharding-jdbc/configuration/config-java.cn.md:293]: explain of KeyGenerateAlgorithm could be removed

File: docs/document/content/manual/sharding-jdbc/configuration/config-java.en.md [hunk 2]
```diff
@@ -279,15 +279,22 @@ The implementation class of `ShardingStrategyConfiguration`, used to configure n
...
-| ----------------- | ---------------------------- | ------------------------------------------------------------------------------------------- |
-| column            | String                       | Column name of key generator                                                                |
-| type              | String                       | Type of key generator, use user-defined ones or built-in ones, e.g. SNOWFLAKE, UUID, LEAF_SEGMENT, LEAF_SNOWFLAKE |
-| props             | Properties                   | The Property configuration of key generators                                                 |
+| *Name*                   | *DataType*               | *Description*                                                |
+| ------------------------ | ------------------------ | ------------------------------------------------------------ |
+| column                   | String                   | Column name of key generator                                 |
+| keyGenerateAlgorithm     | KeyGenerateAlgorithm     | key generate algorithm                                       |
+
+#### KeyGenerateAlgorithm
```
Comment (cherrylzhao) [inline docs/document/content/manual/sharding-jdbc/configuration/config-java.en.md:287]: explain of KeyGenerateAlgorithm could be removed

File: docs/document/content/manual/sharding-jdbc/configuration/config-java.cn.md (hunk 1 above)
Comment (TaoZhiMLND) [reply docs/document/content/manual/sharding-jdbc/configuration/config-java.cn.md]: done

File: docs/document/content/manual/sharding-jdbc/configuration/config-java.en.md (hunk 2 above)
Comment (TaoZhiMLND) [reply docs/document/content/manual/sharding-jdbc/configuration/config-java.en.md]: done

**Pull request closed without merging**